src/test/java/uk/gov/hmcts/darts/log/service/impl/EventLoggerServiceImplTest.java

```diff
@@ -0,0 +1,152 @@
+package uk.gov.hmcts.darts.log.service.impl;
+
+import nl.altindag.log.LogCaptor;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.hmcts.darts.event.model.DartsEvent;
+import uk.gov.hmcts.darts.log.enums.EventSource;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith({MockitoExtension.class})
+@SuppressWarnings("PMD.LawOfDemeter")
+class EventLoggerServiceImplTest {
+
+    EventLoggerServiceImpl eventLoggerService;
+
+    private static LogCaptor logCaptor;
+
+    @BeforeAll
+    public static void setupLogCaptor() {
+        logCaptor = LogCaptor.forClass(EventLoggerServiceImpl.class);
+        logCaptor.setLogLevelToInfo();
+    }
+
+    @AfterEach
+    public void clearLogs() {
+        logCaptor.clearLogs();
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        logCaptor.close();
+    }
+
+    @BeforeEach
+    void setUp() {
+        eventLoggerService = new EventLoggerServiceImpl();
+        ReflectionTestUtils.setField(eventLoggerService, "xhbDailyTestEventText", "Xhibit Daily Test");
+        ReflectionTestUtils.setField(eventLoggerService, "cppDailyTestEventText", "CPP Daily Test");
+        ReflectionTestUtils.setField(eventLoggerService, "xhbEventTypeLength", 5);
+        ReflectionTestUtils.setField(eventLoggerService, "cppEventTypeLength", 4);
```

Comment:
Can we add an `@AllArgsConstructor` to `EventLoggerServiceImpl` to allow the class to be instantiated with desired attribute values? Ideally we should avoid using `ReflectionUtils` as it can be brittle if attribute names are ever refactored.